### DISCUSSION TITLE
GPU parallel computing with CUDA-Fortran

### DISCUSSION THREAD

**Discussion by bibiboom89** [neperfepx/FEPX#104] [Ideas]
Hi,

I'm currently working on a the development of a Finite Element software using parallel computing on GPU using CUDA-Fortran.
This allows a great improvement of the computational time (60 times faster than without parallelization for meshes of 800k 10-nodes tetrahedron elements with a RTX3090). Currently my model is only linear-elastic, but I'm sure it would be possible to do it also for elasto-viscoplastic behaviour.

So my question is : will you consider GPU parallel computing ? Your code being already in fortran, the transition should be easy.

Regards,
Remy

**Comment by mkasemer** [comment]
Hi @bibiboom89 : I believe that you meant to post this in the FEPX discussion forum, am I correct?

If so, if you are interested in helping develop GPU parallelization in FEPX, I would be happy to discuss this with you. Currently we have no plans to do this ourselves, but it is something that were you interested in contributing to the code base, the overhead for you to implement in FEPX is probably less than the overhead for you to extend your code from linear-elastic to full elasto-viscoplastic.

Please let me know, and we can discuss further.

**Reply by bibiboom89** [reply]
Hi @mkasemer,

My bad, I posted in the wrong place. 
Yes, I meant to post in the FEPX discussion forum.

Unfortunately, I don't really have time at the moment to work on this. 
I have to code a solver for heat transfer now along side other projects.
It was more a suggestion than an offer of me working on this ahah.
Maybe one day :)

Regards,
Remy
